code/game/machinery/dna_infuser/organ_sets/carp_organs.dm

```diff
@@ -13,17 +13,21 @@
 
 /datum/status_effect/organ_set_bonus/carp/enable_bonus()
 	. = ..()
-	ADD_TRAIT(src, TRAIT_SPACEWALK, REF(src))
+	ADD_TRAIT(owner, TRAIT_SPACEWALK, REF(src))
 
 /datum/status_effect/organ_set_bonus/carp/disable_bonus()
 	. = ..()
-	REMOVE_TRAIT(src, TRAIT_SPACEWALK, REF(src))
+	REMOVE_TRAIT(owner, TRAIT_SPACEWALK, REF(src))
 
 ///Carp lungs! You can breathe in space! Oh... you can't breathe on the station, you need low oxygen environments.
+/// Inverts behavior of lungs. Bypasses suffocation due to space / lack of gas, but also allows Oxygen to suffocate.
 /obj/item/organ/internal/lungs/carp
 	name = "mutated carp-lungs"
 	desc = "Carp DNA infused into what was once some normal lungs."
-	safe_oxygen_min = 0 //we don't breathe this!
+	// Oxygen causes suffocation.
+	safe_oxygen_min = 0
+	safe_oxygen_max = 1
```

Comment:
Low O2 environments (enough to hurt humans) would start at 15 for the record. 

Up to you if you want to change it, but I personally feel like having the safe_oxygen_max being right at the line where humans would start suffocating due to low O2 levels would be more in line with the theme.